var/spack/repos/builtin/packages/cray-fftw/package.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class CrayFftw(Package):
+    """FFTW is a C subroutine library for computing the discrete Fourier
+       transform (DFT) in one or more dimensions, of arbitrary input
+       size, and of both real and complex data (as well as of even/odd
+       data, i.e. the discrete cosine/sine transforms or DCT/DST).
+       This package is a wrapper for Cray's version of FFTW.
+
+       To install this package, list it as an external package in packages.yaml,
+       and make sure to load the correct cray-fftw module. In some cases you
+       need to load cray-mpich before cray-fftw.
```

Comment:
Curious:  Can these cases be specified as dependencies?

Reply:
I'm afraid it's not being picked up, and we should really fix that. If I add `depends_on('cray-mpich')` and register `cray-mpich` as external + not buildable + specify a module, I'm not getting it to load:

```
$ spack build-env fftw-example@develop %gcc@:8 ^cmake@3.19.1 -- /bin/bash -c 'module list |& grep mpich'
.. nothing ..
```

Also as mentioned above, the order in which modules are loaded is important. It took me a while to figure out that running `module load cray-fftw` and _then_ `module load cray-mpich` causes troubles with GCC because Cray likes to add `-Wl,--as-needed` to all kinds of libs, and [that makes linking order important](https://wiki.gentoo.org/wiki/Project:Quality_Assurance/As-needed#Importance_of_linking_order) :sweat: